src/praisonai-agents/praisonaiagents/memory/memory.py

```
@@ -154,6 +157,31 @@ def _init_protocol_driven_memory(self):
+                        "mongodb": create_mongodb_memory_adapter,
+                    }
+                    if adapter_name in factory_map:
+                        # Call the factory to trigger the ImportError with installation hint
+                        adapter_config = self._get_adapter_config_for_provider(adapter_name)
+                        factory_map[adapter_name](**adapter_config)
+                except ImportError:
+                    # Re-raise the ImportError with installation instructions
+                    raise
+                except Exception:
+                    # Fall through to fallback behavior for other errors
+                    pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)